### PR TITLE
Add planned enhancements to DG and modify known issues in UG

### PR DESCRIPTION
**Pull request by appleshill**
Pls vet thru and make sure i didn't miss anything

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -624,3 +624,20 @@ testers are expected to do more *exploratory* testing.
...
+3. **Accept multiple phone numbers**: Currently, each contact can only have one `PHONE_NUMBER`, which may inconvenience some users who may wish to add contacts with multiple `PHONE_NUMBER`s. We plan to adjust the manner in which  `PHONE_NUMBER`s are stored, using a set similar to the manner in which `Tag` is stored, so that users are able to add more than one `PHONE_NUMBER`.
+4. **Accept more types of characters in phone numbers**: Currently, the `PHONE_NUMBER` field in each contact does not currently allow characters other than `+` and numbers. While as per the ITU-T, additional characters are not a strict necessity for for a `PHONE_NUMBER` to be usable globally, users in certain regions may prefer the inclusion of additional characters to improve readability. Additional characters can include, but are not limited to, ` ` (whitespace), `-`, `(` or `)`. We plan to add support for these additional characters by adjusting the validation RegEX in the Phone class.
```
Comment (valhrd) [inline docs/DeveloperGuide.md:637]: Do we have to define or inform them what this is?

Comment (valhrd) [inline docs/DeveloperGuide.md:637]: Should there be 2 "for"s?

"...additional characters are not a strict necessity for for a `PHONE_NUMBER`..."

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -624,3 +624,20 @@ testers are expected to do more *exploratory* testing.
...
+8. **Accept negative numbers and mathematical expressions in Tag names and values**: Currently, the `TAG` and `TAG:value`'s TagName and TagValue fields do not accept mathematical symbols or expressions, including, but not limited to, `+`, `-`, `*` or `/`, along with negative numbers. This could cause an inconvenience to certain users, who may wish to store `TAG`s with such information. Hence, we intend to adjust the validation RegEx of the `Tag` class to allow for such expressions.
+9. **Block fullscreen of application in Unix and Unix-Like environments**: On certain systems with Unix-based or Unix-like operating systems, including but not limited to MacOS, Linux and its distributions, making the application run in full screen mode leads to slightly buggy behaviour in the GUI, such as the Help window not scaling fully to the size of the display. In This may inconvenience a minority of users who attempt to run the application in fullscreen mode. Hence, we intend to modify the `UiManager` file to block attempts for Unix-based and Unix-like operating systems from making the application run in fullscreen mode.
```
Comment (valhrd) [inline docs/DeveloperGuide.md:642]: "In This may inconvenience..." should not have "In" at the start I believe

Comment (valhrd) [reply docs/DeveloperGuide.md:642]: Unless I am mistaken, I also don't think we should say "minority of users" considering running applications in fullscreen is quite the norm.

We can instead just say "users who attempt to run...".

**Review by valhrd — COMMENTED** [review]
LGTM in general but there are slight grammatical errors and some parts can be improved upon in my opinion.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (appleshill) [reply docs/DeveloperGuide.md:637]: This is the dev guide, and it does not need to have the same target audience as the user guide, so I don't think so. 
Screenshot 2024-11-10 at 9 28 21 PM

**Pull request merged**